common/Services/ComputeSystemService.cs

```diff
@@ -102,4 +103,19 @@ public async Task<List<ComputeSystemProviderDetails>> GetComputeSystemProvidersA
 
         return computeSystemProvidersFromAllExtensions;
     }
+
+    /// <summary>
+    /// Gets a boolean indicating whether the extension should be disabled if the Windows optional feature is absent from the machine.
+    /// </summary>
+    /// <param name="extensionClassId">The class Id of the out of proc extension object</param>
+    /// <returns>True when the Windows optional feature is absent. False otherwise.</returns>
+    private bool ShouldDisabledExtensionIfFeatureAbsent(string extensionClassId)

```

Comment:
```suggestion
    private bool ShouldDisableExtensionIfFeatureAbsent(string extensionClassId)
```

Reply:
I'd propose to move this functionality to ManagementInfrastructureHelper. If we'll have another extensionClassId outside of this file that can be disabled, we'd need to duplicate this code. ManagementInfrastructureHelper can have this ExtensionToFeatureNameMap and answer IsWindowsFeatureAvailable question.

Reply:
updated the method name and added Sergeys suggestion